types/src/key.rs

```diff
@@ -389,6 +390,43 @@ impl From<ContractPackageHash> for Key {
     }
 }
 
+#[derive(Debug, PartialEq, Eq)]
+struct PaddedEraId(u64);
+
+impl PaddedEraId {
+    const SERIALIZED_LENGTH: usize = 32;
+
+    const U64_LE_BYTES_LENGTH: usize = 8;
+
+    const ZEROES_LENGTH: usize = Self::SERIALIZED_LENGTH - Self::U64_LE_BYTES_LENGTH;
+
+    const fn into_inner(self) -> u64 {
+        self.0
+    }
+}
+
+impl ToBytes for PaddedEraId {
+    fn to_bytes(&self) -> Result<Vec<u8>, Error> {
+        let mut buff = Vec::new();
+        buff.extend_from_slice(&[0u8; Self::ZEROES_LENGTH]);
```

Comment:
Heh, actually started this way, but I was having a hell of time getting the `vec!` macro in scope in the wasm builds.  Any tips?

Reply:
I think it needs
```rust
#[macro_use]
extern crate alloc;
```
[The alloc crate is an exception](https://doc.rust-lang.org/edition-guide/rust-2018/module-system/path-clarity.html?highlight=extern#an-exception) to the 2018 edition not generally requiring explicit `extern crate` calls.

Since this is more verbose than what you already have, I'm not fussed if you skip this suggestion :)